src/osp/core/buffer_format.h

```diff
@@ -0,0 +1,139 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2024 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#pragma once
+
+#include <Corrade/Containers/Array.h>
+#include <Corrade/Containers/StridedArrayView.h>
+
+namespace osp
+{
+
+/**
+ * @brief Buffer Attribute Format. Describes how to access element attribute data within a buffer.
+ *
+ * This is useful for SIMD, GPU, and serialization. A SIMD n-body simulation may prefer [XXXYYYZZZ]
+ * to store positions, but GPU mesh vertex positions prefer [XYZXYZXYZ...].
+ */
+template <typename T>
+struct BufAttribFormat
+{
+    using View_t        = Corrade::Containers::StridedArrayView1D<T>;
+    using ViewConst_t   = Corrade::Containers::StridedArrayView1D<T const>;
+    using Data_t        = Corrade::Containers::ArrayView<unsigned char>;
+    using DataConst_t   = Corrade::Containers::ArrayView<unsigned char const>;
+
+    constexpr View_t view(Data_t data, std::size_t count) const noexcept
+    {
+        return stridedArrayView<T>(data, reinterpret_cast<T*>(&data[offset]), count, stride);
+    }
+
+    constexpr ViewConst_t view_const(DataConst_t data, std::size_t count) const noexcept
+    {
+        return stridedArrayView<T const>(data, reinterpret_cast<T const*>(&data[offset]), count, stride);
+    }
+
+    constexpr bool is_not_used() const noexcept { return stride == 0; }
+
+    std::size_t     offset{};
+    std::ptrdiff_t  stride{};
```

Comment:
empty-brace-init instead of = 0 ?

Reply:
I think I mentioned this in another review comment: my intention for an empty brace is "don't care what this value is, just not uninitialized." A zero explicitly indicates that the value is important to some business logic somewhere.

Reply:
oh, it's this one: https://github.com/TheOpenSpaceProgram/osp-magnum/pull/293/files#r1642043900